packages/grid/src/vaadin-grid-selection-column.js

```diff
@@ -149,6 +162,81 @@ class GridSelectionColumn extends GridColumn {
     checkbox.indeterminate = this.__indeterminate;
   }
 
+  /** @private */
+  __lassoAutoScroller() {
+    if (this.__lassoDragStartIndex !== undefined) {
+      // Get the row being hovered over
+      const renderedRows = this._grid._getRenderedRows();
+      let rowHeight = 30;
+      const hoveredRow = renderedRows.find((row) => {
+        const rowRect = row.getBoundingClientRect();
+        rowHeight = rowRect.height;
+        return this.__lassoCurrentY >= rowRect.top && this.__lassoCurrentY <= rowRect.bottom;
+      });
+
+      // Get the index of the row being hovered over or the first/last
+      // visible row if hovering outside the grid
+      let hoveredIndex = hoveredRow ? hoveredRow.index : undefined;
+      const gridRect = this._grid.getBoundingClientRect();
+      if (this.__lassoCurrentY < gridRect.top) {
+        hoveredIndex = this._grid._firstVisibleIndex;
+      } else if (this.__lassoCurrentY > gridRect.bottom) {
+        hoveredIndex = this._grid._lastVisibleIndex;
+      }
+
+      if (hoveredIndex !== undefined) {
+        // Select all items between the start and the current row
+        renderedRows.forEach((row) => {
+          if (
+            (hoveredIndex > this.__lassoDragStartIndex &&
+              row.index >= this.__lassoDragStartIndex &&
+              row.index <= hoveredIndex) ||
+            (hoveredIndex < this.__lassoDragStartIndex &&
+              row.index <= this.__lassoDragStartIndex &&
+              row.index >= hoveredIndex)
+          ) {
+            if (this.__lassoSelect) {
+              this._grid.selectItem(row._item);
+            } else {
+              this._grid.deselectItem(row._item);
+            }
+          }
+        });
+      }
+
+      // Auto scroll the grid
+      this._grid.$.table.scrollTop += (this.__lassoDy || 0) / (rowHeight / 2);
```

Comment:
Scrolling while selecting results in a pretty bad UX because everything is moving all the time. Also quick actions, for example dragging from first to fifth row, will fail, because you end up selecting more than five rows due to the scrolling.

I would suggest to align the implementation with the V8 grid, which only scrolls when the cursor is near the top or bottom of the grid, gradually increasing the scrolled amount the closer to the edge you get: https://demo.vaadin.com/sampler/#ui/grids-and-trees/grid/multi-select

Reply:
Done

Reply:
Was not really happy with the scrolling behavior. Basing it on the hovered row seemed unnecessary, and using the drag delta for the scroll amount made it feel inconsistent. Replaced it with a solution that just scrolls by a fixed amount that increases the closer you get to the grid's edge, which is now very close to how it worked in V8: https://github.com/vaadin/web-components/commit/32a7a9a5e70138a263699da177389a5e1505e516

Reply:
LGTM